### ISSUE TITLE
Support Authentication

### ISSUE THREAD

**Issue by mosheka** [Legion2/swagger-ui-action#431]
We get the following error:
HttpError: API rate limit exceeded for x.x.x.x. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)

It would be great to support user/token method

**Comment by Legion2** [comment]
Which API rate limit is this?

**Comment by mosheka** [comment]
I might have deleted this run.
I think that it was while it tried to retrieve the files.
I will update as soon as it will happen again.

**Comment by Legion2** [comment]
Does this happen during the download of the swagger UI zip from GitHub Release or while querying the GitHub API for the current Releases?

**Comment by mosheka** [comment]
We got it once again:
These are the logs we get:

```
Run Legion[2](https://github.com/-----/webapp/actions/runs/4013379467/jobs/6892655593#step:4:2)/swagger-ui-action@v1
  with:
    output: swagger-ui
    spec-file: src/utils/swagger.yaml
    version: ^[3](https://github.com/-------/webapp/actions/runs/4013379467/jobs/6892655593#step:4:3).0.0
Error: HttpError: API rate limit exceeded for [5](https://github.com/--------/webapp/actions/runs/4013379467/jobs/6892655593#step:4:5)2.22[6](https://github.com/--------/webapp/actions/runs/4013379467/jobs/6892655593#step:4:7).113.33. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
```

**Comment by Legion2** [comment]
Looks like https://github.com/octokit/auth-action.js must be added to the action to allow authentication. This also requires that the GITHUB_TOKEN is explicitly provided to the action.

**Comment by mosheka** [comment]
@Legion2 is it 1) something we need to do (meaning only documentation update is needed) or 2) fix is needed for for the swagger-ui-action?

**Comment by Legion2** [comment]
Both

**Comment by imalao** [comment]
+1 My team is also running into this issue and it'd be great if it can be supported!

**Comment by Legion2** [comment]
I implemented authentication in #464. It uses the github action token to authenticate with the github api and increase the rate limit. @imalao and @mosheka can you please test the change.
